lstchain/image/muon/muon_analysis.py

```diff
@@ -131,90 +131,135 @@ def analyze_muon_event(event_id, image, geom, equivalent_focal_length,
+            muonringparam.ring_center_y,
+            threshold=30,
+            bins=30)
+
+        pix_ringwidth_im = image * dist_ringwidth_mask
+        idx_ringwidth = np.nonzero(pix_ringwidth_im)
+        muonintensityoutput.ring_pix_completeness = npix_above_threshold(
+            pix_ringwidth_im[idx_ringwidth], tailcuts[0]) / len(
+                pix_ring[idx_ringwidth])
+
+    else:
+            muonintensityoutput = MuonIntensityParameter()
+            # Set default values for cases in which the muon intensity fit (with MuonLineIntegrate) is not done:
+            muonintensityoutput.ring_width = np.nan*u.deg
+            muonintensityoutput.impact_parameter_pos_x = np.nan*u.m
+            muonintensityoutput.impact_parameter_pos_y = np.nan*u.m
+            muonintensityoutput.impact_parameter = np.nan*u.m
+            muonintensityoutput.ring_pix_completeness = np.nan
+            muonintensityoutput.ring_completeness = np.nan
+
 
-    muonintensityoutput = ctel.fit_muon(muonringparam.ring_center_x,
-                                    muonringparam.ring_center_y,
-                                    muonringparam.ring_radius,
-                                    x[dist_mask], y[dist_mask],
-                                    image[dist_mask])
     muonintensityoutput.mask = dist_mask
-    idx_ring = np.nonzero(pix_ring)
-    muonintensityoutput.ring_completeness = ring_completeness(
-                    x[idx_ring], y[idx_ring], pix_ring[idx_ring],
-                    muonringparam.ring_radius,
-                    muonringparam.ring_center_x,
-                    muonringparam.ring_center_y,
-                    threshold=30,
-                    bins=30)
     muonintensityoutput.ring_size = np.sum(pix_ring)
-    size_outside_ring = np.sum(pix_out_ring * clean_mask)
-    dist_ringwidth_mask = np.abs(dist - muonringparam.ring_radius
-                                             ) < (muonintensityoutput.ring_width)
-    pix_ringwidth_im = image * dist_ringwidth_mask
-    idx_ringwidth = np.nonzero(pix_ringwidth_im)
-
-    muonintensityoutput.ring_pix_completeness = npix_above_threshold(
-                    pix_ringwidth_im[idx_ringwidth], tailcuts[0]) / len(
-                    pix_ring[idx_ringwidth])
-
-    print("Impact parameter = %s"
-                             "ring_width=%s, ring radius=%s, ring completeness=%s"% (
-                             muonintensityoutput.impact_parameter,
-                             muonintensityoutput.ring_width,
-                             muonringparam.ring_radius,
-                             muonintensityoutput.ring_completeness))
+    size_outside_ring = np.sum(pix_outside_ring * clean_mask)
 
+    # This is just mean charge per pixel in pixels just around the ring (on the outer side):
+    mean_pixel_charge_around_ring = np.sum(pix_ring_2)/len(pix_ring_2) 
+
+    if candidate_clean_ring:
+        print("Impact parameter=%s, ring_width=%s, ring radius=%s, ring completeness=%s"%
```

Comment:
This is a very strange mixture of the very old python string formatting and the newer method. Please replace with:

```
print("Impact parameter={:.3f}, ring_width={:.3f}, ring radius={:.3f}, ring completeness={:.3f}".format(
   muonintensityoutput.impact_parameter, muonintensityoutput.ring_width, 
   muonringparam.ring_radius, muonintensityoutput.ring_completeness,
))
```

Reply:
Ok, thanks, done.